drivers/builtin/src/chacha20.c

```diff
@@ -213,6 +468,37 @@ int mbedtls_chacha20_update(mbedtls_chacha20_context *ctx,
         size--;
     }
 
+#if MBEDTLS_CHACHA20_NEON_MULTIBLOCK != 0
```

Comment:
Do you think we could move the neon code into its own source file? It is easier to read/maintain like that and that is how we have been doing it with AES as well.

(The two implementations seem to be nicely separated, I would hope that this is not a very big change.)